packages/docusaurus-utils-common/src/applyTrailingSlash.ts

```diff
@@ -55,3 +52,27 @@ export default function applyTrailingSlash(
 
   return path.replace(pathname, newPathname);
 }
+
+/** Adds a given string prefix to `str`. */
+export function addPrefix(str: string, prefix: string): string {
+  return str.startsWith(prefix) ? str : `${prefix}${str}`;
+}
+
+/** Removes a given string suffix from `str`. */
+export function removeSuffix(str: string, suffix: string): string {
+  if (suffix === '') {
+    // str.slice(0, 0) is ""
+    return str;
+  }
+  return str.endsWith(suffix) ? str.slice(0, -suffix.length) : str;
+}
```

Comment:
I would move these 2 to a new `stringUtils.ts`, they are not really related to trailing slash but a more general concept